cmd/devp2p/discv5cmd.go

```diff
@@ -136,12 +155,105 @@ func discv5Listen(ctx *cli.Context) error {
 	defer disc.Close()
 
 	fmt.Println(disc.Self())
+
+	if ctx.Bool(discv5DumpFlag.Name) {
+		ticker := time.NewTicker(10 * time.Second)
+		defer ticker.Stop()
+		for range ticker.C {
+			nodes := disc.AllNodes()
+			fmt.Printf("\n--- Dumping %d peers ---\n", len(nodes))
+			for i, n := range nodes {
+				fmt.Printf("\n[%d] %s\n", i+1, n.String())
+				dumpRecord(os.Stdout, n.Record())
+			}
+		}
+	}
+
 	select {}
 }
 
+// opStackENRData is the ENR entry for OP Stack chain identification.
+type opStackENRData struct {
+	chainID uint64
+	version uint64
+}
+
+func (o *opStackENRData) ENRKey() string { return "opstack" }
+
+func (o *opStackENRData) DecodeRLP(s *rlp.Stream) error {
+	b, err := s.Bytes()
+	if err != nil {
+		return fmt.Errorf("failed to decode outer ENR entry: %w", err)
+	}
+	r := bytes.NewReader(b)
+	chainID, err := binary.ReadUvarint(r)
+	if err != nil {
+		return fmt.Errorf("failed to read chain ID var int: %w", err)
+	}
+	version, err := binary.ReadUvarint(r)
+	if err != nil {
+		return fmt.Errorf("failed to read version var int: %w", err)
+	}
+	o.chainID = chainID
+	o.version = version
+	return nil
+}
+
+var _ enr.Entry = (*opStackENRData)(nil)
+
+// chainIDENRData is the ENR entry for chain ID identification.
+type chainIDENRData uint64
+
+func (c chainIDENRData) ENRKey() string { return "chainID" }
+
+var _ enr.Entry = (*chainIDENRData)(nil)
+
 // startV5 starts an ephemeral discovery v5 node.
 func startV5(ctx *cli.Context) (*discover.UDPv5, discover.Config) {
 	ln, config := makeDiscoveryConfig(ctx)
+
+	// Set up OP Stack chain ID filter if specified
+	if ctx.IsSet(opStackChainIDFlag.Name) {
+		expectedChainID := ctx.Uint64(opStackChainIDFlag.Name)
+		config.NodeFilter = func(node *enode.Node) bool {
+			var dat opStackENRData
+			if err := node.Load(&dat); err != nil {
+				log.Debug("Node has no opstack ENR entry", "id", node.ID(), "ip", node.IP(), "err", err)
+				return false
+			}
+			if dat.chainID != expectedChainID {
+				log.Info("Node has different chain ID", "id", node.ID(), "ip", node.IP(), "got", dat.chainID, "expected", expectedChainID)
```

Comment:
It will be nice to change the log level to debug, otherwise there will be a lot logs when running devp2p